File: src/types/face.cpp
#include "face.hpp"

void Face::merge(const Face& other) {
    if (other.fg_) { this->fg_ = other.fg_; }
    if (other.bg_) { this->bg_ = other.bg_; }

    if (other.bold_) { this->bold_ = other.bold_; }
    if (other.italic_) { this->italic_ = other.italic_; }
    if (other.underline_) { this->underline_ = other.underline_; }
    if (other.strikethrough_) { this->strikethrough_ = other.strikethrough_; }
}

File: lua/bindings/face.lua
--- @meta

--- Faces define RGB colors of Cells drawn to the Display.
--- @class Core.Face
--- @field fg Core.Rgb? Foreground color.
--- @field bg Core.Rgb? Background color.
--- @field uc Core.Rgb? Underline color.
--- @field bold boolean
--- @field italic boolean
--- @field underline boolean
--- @field squiggly boolean
--- @field strikethrough boolean
Core.Face = {}

--- @class Core.FaceOptions
--- @field fg Core.Rgb?
--- @field bg Core.Rgb?
--- @field uc Core.Rgb?
--- @field bold boolean?
--- @field italic boolean?
--- @field underline boolean?
--- @field squiggly boolean?
--- @field strikethrough boolean?
local FaceOptions = {}

--- @param options? Core.FaceOptions
--- @return Core.Face
function Core.Face(options) end

--- Clones this object.
--- @return Core.Face
function Core.Face:clone() end
